### DISCUSSION TITLE
Annotating `HashMap<i64, CoolType>` to use `number` instead of `bigint` for Keys

### DISCUSSION THREAD

**Discussion by hongyilyu** [Aleph-Alpha/ts-rs#362] [Q&A] [answered]
Hi there,

I'm working on serializing  `HashMap<i64, CoolType>` using ts_rs. By default, the generated TypeScript type looks like this:
```ts
{ [key in bigint]?: Array<CoolType> }
```
However, I would prefer to use number for the keys instead of bigint. I attempted to annotate the map directly with a custom type, but encountered an issue where CoolType is not imported correctly. Here's what I tried:
```rs
#[derive(Debug, Clone, Serialize, TS)]
#[ts(export)]
pub struct AwesomeStruct {
    #[ts(type = "{ [key in number]?: Array<CoolType> }")]
    pub cool_types: HashMap<i64, Vec<CoolType>>,
}
```

The problem with this approach is that `ts_rs` does not recognize the use of `CoolType`, leading to TypeScript compilation errors.

What is the recommended way to annotate a type within a type?

**Comment by NyxCode** [comment, marked as answer]
Hey!  
I think `#[ts(as = "HashMap<i32, Vec<CoolType>>")]` should work.

**Reply by hongyilyu** [reply]
That works!
Thanks for the response!!!
